### PR TITLE
importer: handle different import media types

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/CERNDocumentServer/cds-ils/issues/921

### CODE REVIEW

File: cds_ils/importer/XMLRecordToJson.py [hunk 1]
```diff
@@ -54,14 +54,18 @@ def dump(self):
             is_deletable = False
 
         init_fields = {}
-        if "im" in marc_record.get("leader", []):
+        if "am" in marc_record.get("leader", []):
+            init_fields.update({"_eitem": {"_type": "e-book"}})
+        elif "im" in marc_record.get("leader", []) or "jm" in marc_record.get(
+            "leader", []
+        ):
```
Comment (kpsherva) [inline cds_ils/importer/XMLRecordToJson.py:61]: small suggestion just for readability
```suggestion
        leader_tag = marc_record.get("leader", [])
        if "am" in leader_tag:
            init_fields.update({"_eitem": {"_type": "e-book"}})
        elif "im" in leader_tag or "jm" in leader_tag:
```
(also in line 68)

File: tests/importer/test_safari_dojson.py [hunk 2]
```diff
@@ -14,14 +15,16 @@ def check_transformation(marcxml_body, json_body):
...
+        init_fields.update({"_eitem": {"_type": "e-book"}})
+    elif "im" in blob.get("leader", []) or "jm" in blob.get("leader", []):
         init_fields.update({"_eitem": {"_type": "audiobook"}})
     elif "gm" in blob.get("leader", []):
         init_fields.update(
             {"document_type": "MULTIMEDIA", "_eitem": {"_type": "video"}}
         )
     else:
-        init_fields.update({"_eitem": {"_type": "e-book"}})
+        raise UnrecognisedImportMediaType(blob.get("leader", []))
```
Comment (kpsherva) [inline tests/importer/test_safari_dojson.py:27]: if it is relatively easy, could you create a test case to catch this exception?

**Comment by ntarocco** [comment]
Is this closing this https://github.com/CERNDocumentServer/cds-ils/issues/914?
If yes, can you add it to the commit?
If not, can you please create an issue and refer to it?

Otherwise it is hard to review.

File: cds_ils/importer/XMLRecordToJson.py [hunk 3]
```diff
@@ -54,14 +54,17 @@ def dump(self):
             is_deletable = False
 
         init_fields = {}
-        if "im" in marc_record.get("leader", []):
+        leader_tag = marc_record.get("leader", [])
+        if "am" in leader_tag:
```
Comment (anikachurilova) [inline cds_ils/importer/XMLRecordToJson.py:58]: This was added to be able to fail with an error if some unknown media type is provided, but to still handle `am` as an `e-book`

File: tests/importer/test_safari_dojson.py [hunk 4]
```diff
@@ -462,6 +467,87 @@ def test_safari_audiobook(app):
         )
 
 
+def test_safari_audiobook2(app):
```
Comment (sakshamarora1) [inline tests/importer/test_safari_dojson.py:470]: Can we please add the leader fields accordingly in description for this test and the test above? It'll make the difference between the tests clearer!

**Review by sakshamarora1 — APPROVED** [review]

**Review by kpsherva — APPROVED** [review]

**Pull request merged**